code-submissions-clustering-core/src/test/kotlin/org/jetbrains/research/code/submissions/clustering/cluster/ClusteringTest.kt

```diff
@@ -194,5 +215,22 @@ class ClusteringTest : ParametrizedBaseWithUnifierTest(getTmpProjectDir()) {
                 ))
             ),
         )
+
+        @JvmStatic
+        fun getTestDataFromBin(): List<Arguments> {
+            val binDir = "${Paths.get("").toAbsolutePath()}/src/test/resources/bin.data"
+            val serializationDirs =
+                File(binDir).list { dir, name -> File(dir, name).isDirectory }.map { Paths.get("$binDir/$it") }
```

Comment:
Please, use [walkTopDown](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.io/java.io.-file/walk-top-down.html) function

Reply:
In this case you can union filtering and toSubmissionsGraph in one call chain